### PR TITLE
Ignore specific url parameters when following location headers

### PR DESCRIPTION
**Pull request by noamblitz**
### Changes

Before, when a sessionid was present in the URL parameters, the location header would also have this sessionid, and the database would blow up with URLs. This code strips certain params from the URL based on the Answer of a Question OOI. Also, some general fixes for questions are present.

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3207
Closes https://github.com/minvws/nl-kat-coordination/issues/3849
Closes https://github.com/minvws/nl-kat-coordination/issues/3848

### Demo
The answer of this question:

image

Makes this config:

image

And when this Location header is present:

image

The sessionid will be stripped so that only this URL is created:

image

### QA notes

Answer the question about oois-in-headers. Find Location headers that have one of the params from the question in them. See that the URLs that are created do not have those params.

---

### Code Checklist

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/tests/integration/test_ooi_deletion.py [hunk 1]
```diff
@@ -131,9 +131,9 @@ def test_events_created_in_worker_during_handling(
         xtdb_octopoes_service.process_event(event)
     xtdb_octopoes_service.commit()
 
-    assert len(event_manager.queue) == 7  # Handling OOI delete event triggers Origin delete event
+    assert len(event_manager.queue) == 8  # Handling OOI delete event triggers Origin delete event
 
-    event = event_manager.queue[6]  # OOID]elete event
+    event = event_manager.queue[7]  # OOID]elete event
```
Comment (underdarknl) [inline octopoes/tests/integration/test_ooi_deletion.py:136]:
```suggestion
    event = event_manager.queue[7]  # OOIDelete event
```

File: octopoes/bits/ask_url_params_to_ignore/question_schema.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$id": "/bit/oois-in-headers",
+  "type": "object",
+  "default": {},
+  "required": [],
```
Comment (underdarknl) [inline octopoes/bits/ask_url_params_to_ignore/question_schema.json:6]:
```suggestion
  "required": ["ignored_url_parameters"],
```

File: octopoes/bits/oois_in_headers/oois_in_headers.py [hunk 3]
```diff
@@ -16,12 +16,31 @@ def is_url(input_str):
     return bool(result.scheme)
 
 
+def get_ignored_url_params(config, config_key, default):
```
Comment (ammar92) [inline octopoes/bits/oois_in_headers/oois_in_headers.py:19]: Missing typing

File: octopoes/bits/oois_in_headers/oois_in_headers.py [hunk 4]
```diff
@@ -16,12 +16,31 @@ def is_url(input_str):
...
+    ignored_url_params = config.get(config_key, None)
+    if ignored_url_params is None:
+        return default
+    return [param.strip() for param in ignored_url_params.split(",")] if ignored_url_params else []
+
+
+def remove_ignored_params(url, ignored_params):
```
Comment (ammar92) [inline octopoes/bits/oois_in_headers/oois_in_headers.py:26]: Missing typing

File: octopoes/bits/oois_in_headers/oois_in_headers.py [hunk 5]
```diff
@@ -16,12 +16,31 @@ def is_url(input_str):
...
+    parsed_url = urlparse(url)
+    query_params = parse_qs(parsed_url.query)
```
Comment (ammar92) [inline octopoes/bits/oois_in_headers/oois_in_headers.py:28]: One potential improvement could be to handle cases where the URL might not have a query string or the query string is empty

```suggestion
    query_params = parse_qs(parsed_url.query)
    if not query_params:
        return url
```

**Review by ammar92 — DISMISSED** [review]
Nice work! Just a few very tiny remarks to improve it even more, but considering the priority I've already approved it

**Review by ammar92 — APPROVED** [review]

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Seems to work. Manually added a HostnameHTTPURL with a 'view' parameter in there. Added the parameter 'view' to the config. Applied the config and the HTTPURL doesn't contain the  'view' parameter. 

### What doesn't work:
The 'removed' parameter doesn't show up on the Config page.

image


### Bug or feature?:
n/a

**Comment by noamblitz** [comment]
I can see "view" in the ignored params of the config in the screenshot

**Review by underdarknl — APPROVED** [review]

**Pull request merged**